cassandra/util.py

```diff
@@ -1193,3 +1193,48 @@ def _sanitize_identifiers(field_names):
                 names_out[index] = "%s_" % (names_out[index],)
             observed_names.add(names_out[index])
     return names_out
+
+
+class Duration(object):
+    """
+    Cassandra Duration Type
+    """
+
+    months = 0
+    days = 0
+    nanoseconds = 0
+
+    def __init__(self, months=0, days=0, nanoseconds=0):
+        self.months = months
+        self.days = days
+        self.nanoseconds = nanoseconds
+        self.validate()
+
+    def validate(self):
```

Comment:
The  server doesn't validate that case. It will rather return a modified Duration with fixed values. Sandeep and I agreed adding this validation on the client could avoid some confusion.
Here is an example of the server behavior without this validation. ie. If you create a Duration(-10,4,2) and fetch it back, you'll get a Duration(-10,-4,-2). 

Let me know if you are fine with it or we can discuss more about that need.